internal/runtime/internal/controller/loader.go

```diff
@@ -807,10 +807,10 @@ func (l *Loader) concurrentEvalFn(n dag.Node, spanCtx context.Context, tracer tr
 	var err error
 	switch n := n.(type) {
 	case BlockNode:
-		ectx := l.cache.GetContext()
 
 		// RLock before evaluate to prevent Evaluating while the config is being reloaded
 		l.mut.RLock()
+		ectx := l.cache.GetContext()
```

Comment:
This looks good, but noticed a few lines down on 826 that defer looks...suspicious. When should that unlock logically happen? If its at 831 would it make sense to wrap that defer in a func call. Feels easy to add logic after later in the function that could be unexpected with that defer.